# Project to simulate a data set using numpy.random

#### The real world phenomenom I have chosen for this project is the gender breakdown at senior management level in Institutes of Technology.   
I will look at the following variables:   
Age   
Qualification   
Years of Experience   
Grade   
Gender      

The age variable will be a non-negative real number between 25 & 66.
The Qualification variable will be a categorical variable with one of 3 different values: Level 8, Level 9, Level 10.
The years of experience variable will be a non-negative real number, and will be based on age and qualification.
The Grade variable will be a categorical variable with one of 7 different values: Assistant Lecturer, Lecturer, Senior Lecturer 1, Senior Lecturer 2, Senior Lecturer 3, Registrar, President
The Gender variable will be either male or female
The determinant of whether a staff member has reached senior management level will be their Grade - those at SL2, SL3, Registrar & President will be considered at Senior Management level
Only staff categorised as Academic will be considered in this project.   


In [1]:
#Import pandas, create columns for dataset.
import pandas as pd
import numpy as np


In [2]:
df = pd.DataFrame(columns=['Age','Gender','Qualification','Years of Experience','Grade'])
df

Empty DataFrame
Columns: [Age, Gender, Qualification, Years of Experience, Grade]
Index: []

In [3]:
#insert age data into dataframe
df ['Age'] = np.random.randint(25, 66, 450)
df


Age Gender Qualification Years of Experience Grade
0     58    NaN           NaN                 NaN   NaN
1     25    NaN           NaN                 NaN   NaN
2     30    NaN           NaN                 NaN   NaN
3     42    NaN           NaN                 NaN   NaN
4     29    NaN           NaN                 NaN   NaN
5     38    NaN           NaN                 NaN   NaN
6     32    NaN           NaN                 NaN   NaN
7     30    NaN           NaN                 NaN   NaN
8     45    NaN           NaN                 NaN   NaN
9     38    NaN           NaN                 NaN   NaN
10    61    NaN           NaN                 NaN   NaN
11    53    NaN           NaN                 NaN   NaN
12    44    NaN           NaN                 NaN   NaN
13    29    NaN           NaN                 NaN   NaN
14    58    NaN           NaN                 NaN   NaN
15    55    NaN           NaN                 NaN   NaN
16    61    NaN           NaN                 NaN   NaN
17    52    NaN           NaN                 NaN   NaN
18    57    NaN           NaN                 NaN   NaN
19    64    NaN           NaN                 NaN   NaN
20    34    NaN           NaN                 NaN   NaN
21    49    NaN           NaN                 NaN   NaN
22    47    NaN           NaN                 NaN   NaN
23    59    NaN           NaN                 NaN   NaN
24    42    NaN           NaN                 NaN   NaN
25    40    NaN           NaN                 NaN   NaN
26    42    NaN           NaN                 NaN   NaN
27    28    NaN           NaN                 NaN   NaN
28    53    NaN           NaN                 NaN   NaN
29    37    NaN           NaN                 NaN   NaN
..   ...    ...           ...                 ...   ...
420   53    NaN           NaN                 NaN   NaN
421   57    NaN           NaN                 NaN   NaN
422   65    NaN           NaN                 NaN   NaN
423   27    NaN           NaN                 NaN   NaN
424   44    NaN           NaN                 NaN   NaN
425   42    NaN           NaN                 NaN   NaN
426   47    NaN           NaN                 NaN   NaN
427   30    NaN           NaN                 NaN   NaN
428   64    NaN           NaN                 NaN   NaN
429   32    NaN           NaN                 NaN   NaN
430   28    NaN           NaN                 NaN   NaN
431   27    NaN           NaN                 NaN   NaN
432   49    NaN           NaN                 NaN   NaN
433   48    NaN           NaN                 NaN   NaN
434   62    NaN           NaN                 NaN   NaN
435   60    NaN           NaN                 NaN   NaN
436   57    NaN           NaN                 NaN   NaN
437   29    NaN           NaN                 NaN   NaN
438   60    NaN           NaN                 NaN   NaN
439   43    NaN           NaN                 NaN   NaN
440   60    NaN           NaN                 NaN   NaN
441   53    NaN           NaN                 NaN   NaN
442   42    NaN           NaN                 NaN   NaN
443   44    NaN           NaN                 NaN   NaN
444   64    NaN           NaN                 NaN   NaN
445   54    NaN           NaN                 NaN   NaN
446   25    NaN           NaN                 NaN   NaN
447   45    NaN           NaN                 NaN   NaN
448   35    NaN           NaN                 NaN   NaN
449   50    NaN           NaN                 NaN   NaN

[450 rows x 5 columns]

In [5]:
#Insert age, gender & qualification columns to dataframe
gender = ['male', 'female']
qual = ['Level 10', 'Level 9', 'Level 8']
df = pd.DataFrame({'Age': np.random.randint(25,66,450),
                   'Gender': np.random.choice(gender, 450,p=[0.4, 0.6]), 
                  'Qualification': np.random.choice(qual, 450, p=[0.21, 0.62, 0.17])})
df

Age  Gender Qualification
0     43  female      Level 10
1     40  female       Level 8
2     64    male       Level 9
3     60    male       Level 9
4     35  female       Level 9
5     50    male       Level 8
6     59  female      Level 10
7     54    male       Level 9
8     40    male      Level 10
9     62  female       Level 9
10    50    male       Level 8
11    36  female       Level 9
12    35  female       Level 8
13    32  female      Level 10
14    34  female       Level 9
15    34  female       Level 9
16    49  female       Level 9
17    32  female      Level 10
18    57  female       Level 9
19    52  female       Level 9
20    51  female       Level 9
21    53  female       Level 9
22    63    male       Level 9
23    61  female      Level 10
24    52  female       Level 9
25    65    male       Level 9
26    37    male       Level 8
27    31    male       Level 9
28    25  female       Level 8
29    60  female       Level 9
..   ...     ...           ...
420   25    male       Level 9
421   31  female      Level 10
422   39    male       Level 9
423   37    male       Level 9
424   54    male       Level 8
425   64  female       Level 9
426   48  female       Level 9
427   26  female      Level 10
428   25  female       Level 9
429   64  female       Level 9
430   51  female       Level 9
431   39    male       Level 9
432   53  female       Level 8
433   32  female       Level 9
434   33    male       Level 9
435   57  female       Level 8
436   65    male       Level 9
437   35    male      Level 10
438   57    male       Level 9
439   35  female       Level 9
440   38    male       Level 9
441   34  female       Level 9
442   52  female       Level 9
443   54    male       Level 9
444   61  female       Level 9
445   45    male       Level 9
446   39  female      Level 10
447   57  female       Level 9
448   40  female       Level 9
449   58  female      Level 10

[450 rows x 3 columns]

After some research into the GMIT staff data I discovered that currently 21% of academic staff hold a PhD, 62% hold a Master's Degree, and the remainder an Honours Degree. I also found that the gender breakdown in academic staff is 40% male, 60% female. After running the above code I realised that I need to look at the relationship between age & qualification, as, for example, a 25 year old staff member is unlikely to hold a PhD, given that the acadmic minimum requirements are a qualification to at least Level 8 plus three years experience. A Level 8 qualification generally takes 3/4 years, a Masters degree another one, and a PhD at least another 3/4, so the minimum age of a PhD holder employed by GMIT in an academic role cannot be less than 27. I couldn't find any recent data on the average age of completion of a PhD, however the median age of a PhD student in Ireland was 26.5 years in 2013, and in 1997 the typical age of a PhD graduate in Ireland was between 24 & 27. 

### References
https://nonpartisaneducation.org/Review/Resources/Int__lHigherEd_AppendixA.pdf   
http://www.iua.ie/wp-content/uploads/2013/08/IUS-01-2013-Report-on-PhD-Students.pdf   
http://hea.ie/assets/uploads/2018/01/Higher-Education-Institutional-Staff-Profiles-by-Gender-2018.pdf
